src/utils/validation/engine.ts

```diff
@@ -0,0 +1,77 @@
+import type { RuleFunction } from "./rules";
+
+export interface FieldConfig {
+    validators?: RuleFunction;
```

Comment:
@anna-liepina  i think this is wrong :( `RuleFunction` is rule, but not a `validator` 

`RuleFunction` returns boolean
while `Validator` should return boolean|string

and may be you didn't noticed typo it is an array :)

Reply:
fixed ^_^